src/client/app/components/IntervalControlsComponent.tsx

```diff
@@ -0,0 +1,169 @@
+	// Special value if custom input for standard menu.
+	const CUSTOM_INPUT = '-99';
+
+	// This is the current interval for the bar and map graphics.
+	const duration = useAppSelector(selectWidthDays);
+	// This is the current compare period for graphic
+	const comparePeriod = chartType === ChartTypes.compare ? useAppSelector(selectComparePeriod) : undefined;
+
+	// Holds the value of standard duration choices used for bar and map, decoupled from custom.
+	const [days, setDays] = React.useState<string>(duration.asDays().toString());
+	// Holds the value during custom duration input for bar and map, so only update graphic
+	// when done entering and separate from standard choices.
+	const [daysCustom, setDaysCustom] = React.useState<number>(duration.asDays());
+	// True if custom duration input for bar or map is active.
+	const [showCustomDuration, setShowCustomDuration] = React.useState<boolean>(false);
+
+	// Keeps react-level state, and redux state in sync.
+	// Two different layers in state may differ especially when externally updated (chart link, history buttons.)
+	React.useEffect(() => {
+		// Assume value is valid since it is coming from state.
+		// Do not allow bad values in state.
+		const isCustom = !(['1', '7', '28'].find(days => days == duration.asDays().toString()));
+		setShowCustomDuration(isCustom);
+		setDaysCustom(duration.asDays());
+		setDays(isCustom ? CUSTOM_INPUT : duration.asDays().toString());
+	}, [duration]);
+
+	// Returns true if this is a valid duration.
+	const daysValid = (days: number) => {
+		return Number.isInteger(days) && days >= MIN_DAYS && days <= MAX_DAYS;
+	};
+
+	// Updates values when the standard duration menu for bar or map is used.
+	const handleDaysChange = (value: string) => {
+		if (value === CUSTOM_INPUT) {
+			// Set menu value from standard value to special value to show custom
+			// and show the custom input area.
+			setDays(CUSTOM_INPUT);
+			setShowCustomDuration(true);
+		} else {
+			// Set the standard menu value, hide the custom duration input
+			// and duration for graphing.
+			// Since controlled values know it is a valid integer.
+			setShowCustomDuration(false);
+			updateDurationChange(Number(value));
+		}
+	};
+
+	// Updates value when the custom duration input is used for bar or map.
+	const handleCustomDaysChange = (value: number) => {
+		setDaysCustom(value);
+	};
+
+	const handleEnter = (key: string) => {
+		// This detects the enter key and then uses the previously entered custom
+		// duration to set the duration for the graphic.
+		if (key === 'Enter') {
+			updateDurationChange(daysCustom);
+		}
+	};
+
+	const updateDurationChange = (value: number) => {
+		// Update if okay value. May not be okay if this came from user entry in custom form.
+		if (daysValid(value)) {
+			dispatch(graphSlice.actions.updateDuration(moment.duration(value, 'days')));
+		}
+	};
+
+	// Define the initial set of options for the dropdown when the chart type is not 'compare'.
```

Comment:
This is definately better than the previous code. However, the custom option is showing on the compare graphic (but it does nothing). What I'm thinking is to add the custom option here and then, below, where you reset the menu for compare it would not be added. I'm open to other ideas but that may get the two set of choices without much effort.

Now, for the future, I'm noting that when we add compare to the custom choices, it would be best to fuse the options so there is only one enum that serves them all.